client/ayon_resolve/plugins/create/create_workfile.py

```diff
@@ -17,8 +22,61 @@ class CreateWorkfile(AutoCreator):
 
     default_variant = "Main"
 
-    def collect_instances(self):
+    def _dumps_data_as_marker(self, data):
+        """Store workfile as timeline marker.
+
+        Args:
+            data (dict): The data to store on the timeline.
+        """
+        # Append global project 
+        # (timeline metadata is not maintained by Resolve native OTIO)
+        timeline = lib.get_current_timeline()
+        timeline_settings = timeline.GetSetting()
+
+        try:
+            pixel_aspect = int(timeline_settings["timelinePixelAspectRatio"])
+        except ValueError:
+            pixel_aspect = 1.0
+
+        data.update({
+            "width": timeline_settings["timelineResolutionWidth"],
+            "height": timeline_settings["timelineResolutionHeight"],
+            "pixelAspect": pixel_aspect
+        })
+
+        # Store as marker note data
+        note = json.dumps(data)
+
+        timeline.AddMarker(
+            timeline.GetStartFrame(),
+            constants.ayon_marker_color,
+            constants.ayon_marker_name,
+            note,
+            constants.ayon_marker_duration
+        )
```

Comment:
@jakubjezek001 Unfortunately it does not look like Resolve allow to register custom metadata at Timeline/Track level (plus Metadata gets totally ignored by the native OTIO exporter right now).

Best solution I could think of was to store the serialized workfile instance along timeline resolution metadata as a marker on the first frame of the timeline itself. 
![image](https://github.com/user-attachments/assets/363754e6-147d-46ba-89e3-7984bda1d1b8)
 
 Is that a good idea ? Do you have a better idea/recommendation ?

Reply:
In all hosts as far as I know the "Workfile" represents the full scene file of the host application. In this case it looks like we're considering only one timeline, and that is the _active_ timeline. Which may be odd behavior? Do we need to store this data PER timeline? Or do we need it once in the full project. If the latter, then...

There doesn't seem to be any API to set `Project Notes` otherwise that could've been handy to put a JSON string.
What if we make a dedicated "Disabled" timeline that we solely use to store the metadata.

![image](https://github.com/user-attachments/assets/786fc097-d6d6-4eee-ba19-e0af3f0bd219)

Reply:
I'm with @BigRoy on this. Instead of the current approach, we could create a sidecar .JSON file in the work folder alongside the .DRP files. This would serve as our go-to for reading and storing metadata. As for clip-level data, we can keep using markers like we do now.

Reply:
Sidecar files have a lot of issues in management - I rarely feel that it's worth it going that route.

It would really be much nicer if we can find a _hacky_ way to store it _within_ the resolve project.

For what it's worth - it may be worth it for you all to make an account on Blackmagic Design forum and just respond to [this topic](https://forum.blackmagicdesign.com/viewtopic.php?f=21&t=189685&hilit=python+database#p987221) to get some visibility on that by Resolve devs, etc. plus maybe the topic already has some other hints. (Ignore the guy who created that topic!)

And look at [this hack mentioned there](https://forum.blackmagicdesign.com/viewtopic.php?f=21&t=189685&hilit=python+database#p991541). Could that suit our needs?

Reply:
> And look at [this hack mentioned there](https://forum.blackmagicdesign.com/viewtopic.php?f=21&t=189685&hilit=python+database#p991541). Could that suit our needs?

Yeah! This is definitely the better way. It actually got the attention in API doc but since there were not list of available  keys and value type. Thanks for the tip @BigRoy